src/gateway/GasService.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {Auth} from "src/Auth.sol";
+import {IGasService} from "src/interfaces/gateway/IGasService.sol";
+import {MathLib} from "src/libraries/MathLib.sol";
+import {MessagesLib} from "src/libraries/MessagesLib.sol";
+
+contract GasService is IGasService, Auth {
+    using MathLib for uint256;
+
+    /// @inheritdoc IGasService
+    uint256 public price;
+    /// @inheritdoc IGasService
+    uint256 public messageCost;
+    /// @inheritdoc IGasService
+    uint256 public proofCost;
+
+    constructor(uint256 messageCost_, uint256 proofCost_, uint256 price_) {
```

Comment:
nit: would put `price` first since it's also first in the variable list above. Also personally I think aesthetically it is always nice to order arguments by shortest to longest (if there is no other semantic reason for the ordering), but that is a super nit ;)